### ISSUE TITLE
Profiles

### ISSUE THREAD

**Issue by tschmidtb51** [secvisogram/secvisogram#145]
CSAF 2.0 introduced with oasis-tcs/csaf#276 profiles. Secvisogram should support them. We need to specify what is actually expected.

**Comment by tschmidtb51** [comment]
CSAF 2.0 specifies in [4 Profiles](https://github.com/oasis-tcs/csaf/blob/master/csaf_2.0/prose/csaf-v2-editor-draft.md#4-profiles) the purpose and requirements of profiles. Now there are basically five options how we can support them in Secvisogram:

### 1. Simple solution: Validation only
- The `/document/category` is not prefilled but provides the options:
  - `generic_csaf`
  - `security_incident_response`
  - `informational_advisory`
  - `security_advisory`
  - `vex`
  - User can fill-in any value
- No additional adding or deletion of fields when the value of `/document/category` changes.
- Secvisogram presents always the minimal valid document (from Generic CSAF).

### 2. Extended solution A: Context-aware addition of fields
- like solution 1 but adds required fields when the value of `/document/category` changes
- does not delete fields when the value of `/document/category` changes

### 3. Extended solution B: Context-aware addition and warned deletion of fields
- like solution 2 but adds a (modal) warning if the changed value would require the deletion of fields. The user is presented a list of fields which would be deleted and a warning about possible data loss. Two options are presented:
  - Do the change: Sets the newly selected value for `/document/category` and deletes the fields as required.
  - Cancel: Keeps the old value for `/document/category` and does not change anything else.

### 4. Full solution: user pre-selects the profile
- like solution 3 but whenever a new (minimal) document is generated, the user must select to use one of the profiles first before getting a new document (default profile and pre-selected: "Generic CSAF"). Each profile needs to have a short explanation from the standard to guide the user which one to use (maybe even link to the corresponding subsection in the standard?)
- In the start-up a minimal document with the profile "Generic CSAF" is loaded

### 5. Advanced solution: Force user to select profile
- like solution 4 but also forces user in the start-up process to select a profile

@sthagen, @pdamian: Which one would you prefer as a user?

**Comment by pdamian** [comment]
I guess I would prefer solution 3 over solutions 1 and 2. I think it would be useful when the user gets informed, about what fields will be removed, when switching a profile.

Regarding solutions 4 and 5: I think it would be good if the user can pre-select a profile and gets some advice on which profile fits best his needs. I would however probably not force the user to do so. Even when pre-selecting a profile, in my opinion it should be possible to later on switch to another one.

**Comment by sthagen** [comment]
Ontologies are easy until reality shows up … with profiles I often ask if the relationship is identity or possession.
When profiles lead to removal of attributes the composition becomes even more difficult.

> Am I a musician or do I just carry a guitar? And, if I am a musician, can I still carry a hammer?

So, as users, if the profiles (including the unadorned core labeled generic CSAF) are mutually exclusive, we may also want the side loading (per copy-pasta into the JSON editor) as well as loading drafts from other sources _work_: 
> How does the application react on these events? 
> Can we detect the profile from the content already present?

Deletion of content with modal dialogs may become difficult to resolve as user, because seeing the list of such conflicts and preparing a mitigation course of action might suggest an assistant to be less annoying. Do we have transforms we can offer to transfer content already entered from one place in the tree to another?

To me, solutions 4 and 5 may become a challenge in the side loading case and when wanting to transform existing content of one profile into content of another profile. Say, a CSAF document starts generic, than the next incarnation is an incident response, then the next is a security advisory, and so on.

Solution 2 should be a no hassle way to implement an initial profile aware assisting technology. Solution 3 may need a lot of effort for corner cases and may also cause irritation or even frustration for the users.

In conclusion I suggest to evolve in the order:

1. Solution 1
2. Solution 2
3. Any of 5, 4, or 3 depending on feedback from 2. by users and implementation experience.

**Comment by bentolor** [comment]
In regard to "Solution 1": So the proposed way is to enhance the current textfield in the Form Editor to support value suggestions for the mentioned 5 values? The field should still accept other custom values. But nothing else needs to be changed in the frontend, as everything else is handled by the validation described in #36 ?

**Comment by tschmidtb51** [comment]
That understanding is correct.

**Comment by tschmidtb51** [comment]
@bentolor: Please implement solution 1.
